tests/unit/api/test_targets.py

```diff
@@ -50,6 +50,65 @@ def test_post(self, monkeypatch, test_client, token_headers):
             },
             "message": "Target(s) successfully submitted.",
         }
+        assert mocked_repository_metadata.apply_async.calls == [
+            pretend.call(
+                kwargs={
+                    "action": "add_targets",
+                    "payload": payload,
+                },
+                task_id=fake_task_id,
+                queue="metadata_repository",
+                acks_late=True,
+            )
+        ]
+
+    def test_post_with_add_task_id_to_custom(
```

Comment:
Why do we need this test when by you already test the case when `add_task_id_to_custom` is true in `test_post`?
Either rename `test_post` or give a comment that in this case we are using `add_task_id_to_custom` or make pass `add_task_id_to_custom` directly in the input `payload`.
Then you can add a simple test case only when `add_task_id_to_custom` is `False`. 

I am not a fan of a single test checking multiple scenarios as we did here.
We can have multiple tests for each case of `add_task_id_to_custom` as we usually do and if there is a repeating code we can always create helper setup functions.

Reply:
> I am not a fan of a single test checking multiple scenarios as we did here. We can have multiple tests for each case of `add_task_id_to_custom` as we usually do and if there is a repeating code we can always create helper setup functions.

In the RSTUF, we are adopting two standards for UT across all components
- One test function per scenario
- Be explicit about the test steps, avoiding nested functions to build scenarios or conditional helpers that will require tests and can bring false positives.